sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -120,7 +120,7 @@ public void testIntermediateFetch() throws Exception {
     public void testHavingGlobalAggregation() throws Exception {
         LogicalPlan plan = plan("select min(a), min(x) from t1 having min(x) < 33 and max(x) > 100");
         assertThat(plan, isPlan("FetchOrEval[min(a), min(x)]\n" +
-                                "Filter[((min(x) < 33) AND (max(x) > 100))]\n" +
+                                "Filter[((cast(min(x) AS long) < 33) AND (cast(max(x) AS long) > 100))]\n" +
```

Comment:
I wonder if we also need some kind of precedence within the symbols to decide which casts are "better". In this case I think it would be preferable to change the literals instead. (From a perspective of what the user probably intended, in this case it shouldn't matter from a performance perspective)

(I think it's fine this way for 2.3, but we might want to tune this in the future)

Reply:
I totally agree, as of now the precedence is strictly checked here and long happens to precede integer, but when we have Literals we could go a different path most of the time by performing a downcast (casting 100 to integer).

Reply:
To add, the downcast already happens as a last resort when the precedence can't be matched due to cast restrictions (i.e. columns).